dipy/viz/ui.py

```diff
@@ -0,0 +1,280 @@
+        """
+        ren.add(*self.get_actors())
+
+        # Get a hold on the current interactor style.
+        iren = ren.GetRenderWindow().GetInteractor().GetInteractorStyle()
+
+        for callback in self._callbacks:
+            if not isinstance(iren, CustomInteractorStyle):
+                msg = ("The ShowManager requires `CustomInteractorStyle` in"
+                       " order to use callbacks.")
+                raise TypeError(msg)
+
+            iren.add_callback(*callback, args=[self])
+
+    def add_callback(self, prop, event_type, callback, priority=0):
+        """ Adds a callback to a specific event for this UI component.
+        Parameters
+        ----------
+        prop : vtkProp
+        event_type : event code
+        callback : function
+        priority : int
+        """
+        # Actually since we need an interactor style we will add the callback
+        # only when this UI component is added to the renderer.
+        self._callbacks.append((prop, event_type, callback, priority))
+
+    def set_center(self, position):
+        """ Sets the center of the UI component
+        Parameters
+        ----------
+        position : (float, float)
+        """
+        msg = "Subclasses of UI must implement `set_center(self, position)`."
+        raise NotImplementedError(msg)
+
+    def set_visibility(self, visibility):
+        """ Sets visibility of this UI component. """
+        for actor in self.get_actors():
+            actor.SetVisibility(visibility)
+
+
+class Button2D(UI):
+    """A 2D overlay button and is of type vtkTexturedActor2D.
+    Currently supports:
+    - Multiple icons.
+    - Switching between icons.
+    """
+
+    def __init__(self, icon_fnames, size=(30, 30)):
+        """
+        Parameters
+        ----------
+        size : 2-tuple of int, optional
+            Button size.
+        icon_fnames : dict
+            {iconname : filename, iconname : filename, ...}
+        """
+        super(Button2D, self).__init__()
+        self.icon_extents = dict()
+        self.icons = self.build_icons(icon_fnames)
+        self.icon_names = list(self.icons.keys())
+        self.current_icon_id = 0
+        self.current_icon_name = self.icon_names[self.current_icon_id]
+        self.actor = self.build_actor(self.icons[self.current_icon_name])
+        self.size = size
+
+    def build_icons(self, icon_fnames):
+        """ Converts file names to vtkImageDataGeometryFilters.
+        A pre-processing step to prevent re-read of file names during every state change.
+        Parameters
+        ----------
+        icon_fnames : dict
+            {iconname: filename, iconname: filename, ...}
+        Returns
+        -------
+        icons : dict
+            A dictionary of corresponding vtkImageDataGeometryFilters
+        """
+        icons = {}
+        for icon_name, icon_fname in icon_fnames.items():
+            png = vtk.vtkPNGReader()
+            png.SetFileName(icon_fname)
```

Comment:
You should check icon filename extension, then ignore and warn the users when it is the wrong extension.
Some people like svg format for icon and vtk generate a segmentation fault in this case.